diff --git a/dist/girder.es.js b/dist/girder.es.js
index bb491287c4008cb3cb54608deb7a0d71314928d7..bacb64835be27b27595206fe26987cdf9fdd7679 100644
--- a/dist/girder.es.js
+++ b/dist/girder.es.js
@@ -4020,7 +4020,7 @@ function j4({ girderRest: e, onFilesChanged: t, onError: n, onDone: r }) {
     if (b.status === "done")
       return A.then(() => b.result);
     const w = (S) => {
-      b.progress = S;
+      Object.assign(b.progress, S);
     };
     return b.status = "uploading", b.progress.indeterminate = !0, b.upload ? A = A.then(() => b.upload.resume()) : (b.upload = new x(b.file, {
       $rest: e,
@@ -17573,7 +17573,7 @@ const xb = /* @__PURE__ */ Nt(q9, [["render", W9], ["__scopeId", "data-v-8623da7
       onDone: (A) => t.emit("done", A)
     }), m = B(
       () => e.multiple ? "Drag files here or click to select them" : "Drag a file here or click to select one"
-    ), y = B(() => a ? `${zr(s)} / ${zr(d)} (${zr(f)}%)` : `${r.length} selected (${zr(d)} total)`);
+    ), y = B(() => a.value ? `${zr(s.value)} / ${zr(d.value)} (${Math.round(f.value)}%)` : `${r.value.length} selected (${zr(d.value)} total)`);
     function x() {
       b({
         dest: e.dest,
@@ -17618,10 +17618,10 @@ function Y9(e, t, n, r, a, i) {
             r.uploading ? (he(), at("div", Z9, pt(r.statusMessage), 1)) : $e("", !0),
             r.uploading ? (he(), Ae(bp, {
               key: 2,
-              value: r.totalProgressPercent,
+              "model-value": r.totalProgressPercent,
               indeterminate: r.indeterminate,
               height: "20"
-            }, null, 8, ["value", "indeterminate"])) : $e("", !0)
+            }, null, 8, ["model-value", "indeterminate"])) : $e("", !0)
           ]),
           _: 1
         })
diff --git a/src/components/Upload/Upload.vue b/src/components/Upload/Upload.vue
index 92a607afbe3b090ead78a29e73022f15b1afa0d7..22d687db8f1a4046fa42ccee3664310fa613c5a3 100644
--- a/src/components/Upload/Upload.vue
+++ b/src/components/Upload/Upload.vue
@@ -61,10 +61,10 @@ export default {
     );
 
     const statusMessage = computed(() => {
-      if (uploading) {
-        return `${formatSize(totalProgress)} / ${formatSize(totalSize)} (${formatSize(totalProgressPercent)}%)`;
+      if (uploading.value) {
+        return `${formatSize(totalProgress.value)} / ${formatSize(totalSize.value)} (${Math.round(totalProgressPercent.value)}%)`;
       }
-      return `${files.length} selected (${formatSize(totalSize)} total)`;
+      return `${files.value.length} selected (${formatSize(totalSize.value)} total)`;
     });
 
     function startUpload() {
@@ -108,7 +108,7 @@ export default {
         </div>
         <v-progress-linear
           v-if="uploading"
-          :value="totalProgressPercent"
+          :model-value="totalProgressPercent"
           :indeterminate="indeterminate"
           height="20"
         />
diff --git a/src/composables/upload.js b/src/composables/upload.js
index 098d7751c46f1fc795c404f50b3ba83e2cabcb41..a3afbe91cdc96941ab2a4e5c4031013f49b2e578 100644
--- a/src/composables/upload.js
+++ b/src/composables/upload.js
@@ -45,7 +45,7 @@ export default function usefileUploader({ girderRest, onFilesChanged, onError, o
     let chain = Promise.resolve();
     if (file.status === 'done') {return chain.then(() => file.result);}
 
-    const progress = (event) => { file.progress = event; };
+    const progress = (event) => { Object.assign(file.progress, event); };
     file.status = 'uploading';
     file.progress.indeterminate = true;
 
